server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileReset.java

```diff
@@ -54,19 +55,22 @@
   private final QProfileFactory factory;
   private final RuleActivator activator;
   private final ActiveRuleIndexer activeRuleIndexer;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
   private final ProfileDefinition[] definitions;
 
-  public QProfileReset(DbClient db, RuleActivator activator, ActiveRuleIndexer activeRuleIndexer, QProfileFactory factory,
-    ProfileDefinition[] definitions) {
+  public QProfileReset(DbClient db, RuleActivator activator, ActiveRuleIndexer activeRuleIndexer, QProfileFactory factory, DefaultOrganizationProvider defaultOrganizationProvider,
```

Comment:
if you replace the array parameter by a vararg parameter, I think the constructor without any `ProfileDefinition` parameter isn't required anymore.
this other constructor is required for pico to be able to instantiate the project when there is no `ProfileDefinition` instance in the container